src/Auditor.ts

```diff
@@ -0,0 +1,99 @@
+import { Address } from '@graphprotocol/graph-ts';
+import {
+  AdjustFactorSet as AdjustFactorSetEvent,
+  LiquidationIncentiveSet as LiquidationIncentiveSetEvent,
+  MarketEntered as MarketEnteredEvent,
+  MarketExited as MarketExitedEvent,
+  MarketListed as MarketListedEvent,
+  PriceFeedSet as PriceFeedSetEvent,
+} from '../generated/Auditor/Auditor';
+import {
+  MarketExit, MarketEnter, MarketList, AdjustFactorSet, LiquidationIncentiveSet, PriceFeedSet,
+} from '../generated/schema';
+import { ERC20 as ERC20Contract } from '../generated/Auditor/ERC20';
+import { Market as MarketContract } from '../generated/Auditor/Market';
+import loadMarket from './utils/loadMarket';
+import loadAccount from './utils/loadAccount';
+import saveMarketState from './utils/saveMarketState';
+
+export function handleMarketListed(event: MarketListedEvent): void {
+  const marketList = new MarketList(event.transaction.hash.toHex());
```

Comment:
I comment on this line as an example, but It applies to all  `event.transaction.hash.toHex()`.
According to the [docs](https://thegraph.com/docs/en/developing/creating-a-subgraph/#recommended-ids-for-creating-new-entities), `event.transaction.hash.toHex()` might not always be unique.
The recommended approach is to `use event.transaction.hash.toHex() + "-" + event.logIndex.toString()`. This ensures uniqueness, even when multiple events are emitted within a single transaction, which might occur during operations such as rollovers or leverage.

Reply:
nice catch, thanks! we have a util function for that (`toId(event)`) that should be used here